File: app/src/main/java/com/example/dcmtkdemo/viewmodel/PacsViewModel.java
package com.example.dcmtkdemo.viewmodel;

import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.example.dcmtkdemo.model.PatientRecord;

import java.util.ArrayList;
import java.util.List;

public class PacsViewModel extends ViewModel {
    public final MutableLiveData<String> host = new MutableLiveData<>("192.168.10.153");
    public final MutableLiveData<Integer> port = new MutableLiveData<>(11112);
    public final MutableLiveData<String> localAet = new MutableLiveData<>("ANDROID_SCU");
    public final MutableLiveData<String> remoteAet = new MutableLiveData<>("ACME_STORE");

    public final MutableLiveData<Boolean> assetsReady = new MutableLiveData<>(false);

    public final MutableLiveData<List<PatientRecord>> queryResults = new MutableLiveData<>(new ArrayList<>());
}

File: dcmtk/src/main/java/com/example/dcmtk/viewmodel/PacsViewModel.kt
package com.example.dcmtk.viewmodel

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.MutableLiveData
import com.example.dcmtk.model.PacsConfig
import com.example.dcmtk.model.PatientRecord
import com.example.dcmtk.utils.PacsPrefs

class PacsViewModel(application: Application) : AndroidViewModel(application) {
    val pacsConfig = MutableLiveData(PacsPrefs.getConfig(application))
    val worklistConfig = MutableLiveData(PacsPrefs.getWorklistConfig(application))

    // Backwards compatibility or convenience getters
    val host get() = pacsConfig.value?.host ?: ""
    val port get() = pacsConfig.value?.port ?: 0
    val localAet get() = pacsConfig.value?.localAet ?: ""
    val remoteAet get() = pacsConfig.value?.remoteAet ?: ""

    fun updateConfig(host: String, port: Int, localAet: String, remoteAet: String) {
        pacsConfig.value = PacsConfig(host, port, localAet, remoteAet)
    }

    val isPacsConnected = MutableLiveData(false)
    val isCEchoSuccess = MutableLiveData(false)

    val assetsReady = MutableLiveData(false)

    val queryResults = MutableLiveData<List<PatientRecord>>(ArrayList())
    val mwlResults = MutableLiveData<List<PatientRecord>>(ArrayList())
}
